.github/workflows/test_pages_build.yaml

```diff
@@ -31,13 +31,34 @@ jobs:
       - name: Install dependencies
         run: poetry install -E docs
 
-      - name: Build documentation
+      - name: Derive files from sources
+        # Note: First, we replace the "0.0.0" placeholder version number in `pyproject.toml`.
+        # Reference: https://github.com/mtkennerly/poetry-dynamic-versioning/blob/master/README.md#command-line-mode
         run: |
-          mkdir -p site
-          touch site/.nojekyll
+          poetry self add "poetry-dynamic-versioning[plugin]"
+          poetry dynamic-versioning
+          make squeaky-clean all
+
+      - name: Generate web-based documentation
+        run: |
+          mkdir -p docs
+          touch docs/.nojekyll
           make gendoc
           poetry run mkdocs build -d site
```

Comment:
I noticed that this command is present in this `test_pages_build.yaml` file, but not in the `deploy-docs.yaml` file. I don't know why it's necessary in one situation, but not the other. Maybe it's because the author wanted the documentation website's file tree to  also be generated in a directory named `site` (instead of—or in addition to—the default directory).